src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts

```diff
@@ -113,7 +114,33 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
   }
 
   currentPageIsNotValid(): boolean {
-    return !this.pageValidationService.isPageValid(this.currentPage, this.editForm);
+    const data = this.editForm.controls.data['controls'];
+    if (data) {
+      if (this.currentPage.id === 'editAppealnewMatters') {
```

Comment:
This is _super_ specific and means you're only fixing it for a single event trigger. If there's another event with the same structure - or they rename the event - this code won't be called. I don't think the toolkit should have handling for specific events like this.

Reply:
All of the calls to `newMatters` and `hasNewMatters` below are also specific to this very one trigger event.